### PR TITLE
VT-6099 WooCommerceNet: Updated solution

### PR DESCRIPTION
**Pull request by mrassudishkin**
# Description

Tickets: [VT-6099]

I know that you see a lot of changes but I made only next here:

- Added solution
- Added empty project for tests
- That's all

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Functionality change (fix or feature that would cause existing functionality to work differently than before)
- [ ] Configuration change
- [ ] New / updated script
- [X] Refactoring
- [ ] New tests to existing code

# PR Readiness Checklist

- [X] Followed [development conventions][1] to the best of my ability
- [X] Code is documented, particularly public interfaces and hard-to-understand areas
- [X] Tests are updated / added and all pass
- [X] Performed a self-review of my own code
- [X] Acceptance criterias are confirmed by testing changes in UI (or another appropriate way)
- [X] Confluence documentation is updated, if needed
- [X] New code does not generate new warnings

[1]: https://agileharbor.atlassian.net/wiki/spaces/DEV/pages/1114130/Conventions

[VT-6099]: https://agileharbor.atlassian.net/browse/VT-6099?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by andrey-kozlov-skuvault — COMMENTED** [review]
@mrassudishkin I think it is better to apply this template - https://github.com/skuvault-integrations/sv-integrations-template to the repo. As it does not have our current structure, as well as some infrastructure files, and their content is not actual.
So I think we can use this template and place all the files there.

**Comment by mrassudishkin** [comment]
> @mrassudishkin I think it is better to apply this template - https://github.com/skuvault-integrations/sv-integrations-template to the repo. As it does not have our current structure, as well as some infrastructure files, and their content is not actual. So I think we can use this template and place all the files there.

Done

File: src/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<Project>
+	<PropertyGroup>
+		<TargetFramework>net6.0</TargetFramework>
```
Comment (dk-skuvault) [inline src/Directory.Build.props:3]: Should it be net6.0?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		<FileVersion>1.0.0.0</FileVersion>
+		<Version>1.0.0</Version>
+		<Authors>SkuVault</Authors>
+		<Company>SkuVault, Inc.</Company>
+		<Copyright>Copyright © 2023 SkuVault, Inc.</Copyright>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<CodeAnalysisTreatWarningsAsErrors>false</CodeAnalysisTreatWarningsAsErrors>
+		<TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (dk-skuvault) [inline src/Directory.Build.props:23]: Can we enable it or we have a lot of such warnings?

File: src/WooCommerce.NET/Base/BaseObject.cs [hunk 3]
```diff
@@ -56,12 +56,12 @@ void OnDeserialized(StreamingContext ctx)
                     {
                         object value = objValue.GetValue(this);
 
-                        if (!(value == null || value.ToString() == string.Empty))
-                        {
-                            if (decimal.TryParse(value.ToString(), NumberStyles.Any, CultureInfo.InvariantCulture, out decimal weight))
-                            {
-                                pi.SetValue(this, weight);
-                            }
+                        if (!(value == null || value.ToString() == string.Empty))
```
Comment (dk-skuvault) [inline src/WooCommerce.NET/Base/BaseObject.cs:59]: Is this whole project formatted with spaces?

File: src/Directory.Build.props (hunk 1 above)
Comment (mrassudishkin) [reply src/Directory.Build.props:3]: I don't change this project, can you please don't force me overwrite all this project. I just wanted make it similar on our other projects and add tests. If you want that I make refactoring then please create new ticket for it. This ticket is about bug.

File: src/Directory.Build.props (hunk 2 above)
Comment (mrassudishkin) [reply src/Directory.Build.props:23]: answered in previous comment

File: src/WooCommerce.NET/Base/BaseObject.cs (hunk 3 above)
Comment (mrassudishkin) [reply src/WooCommerce.NET/Base/BaseObject.cs:59]: Yes. answered in previous comment

**Review by andrey-kozlov-skuvault — APPROVED** [review]

**Review by dk-skuvault — DISMISSED** [review]

File: src/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		<EnableNETAnalyzers>true</EnableNETAnalyzers>
+		<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+		<AnalysisMode>Recommended</AnalysisMode>
+		<AnalysisLevel>latest</AnalysisLevel>
+		<LangVersion>latest</LangVersion>
+		<Platforms>x64</Platforms>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<AssemblyVersion>1.0.0.0</AssemblyVersion>
```
Comment (andrey-kozlov-skuvault) [inline src/Directory.Build.props:13]: Does this reflect correct version for this project?

**Review by dk-skuvault — APPROVED** [review]

**Review by andrey-kozlov-skuvault — APPROVED** [review]

**Pull request merged**